cumulusci/plugins/append_where_clause.py

```diff
@@ -0,0 +1,16 @@
+def append_where_clause(soql, object_config):
+    """Function that applies where clause to soql if it is defined in mapping yml file"""
+
```

Comment:
The `soql` input for this function might already have a WHERE clause if the mapping specifies a record type (see https://github.com/SFDO-Tooling/CumulusCI/blob/main/cumulusci/tasks/bulkdata/extract.py#L139). We need to join the two together with AND if the mapping specifies both `record_type` and `soql_filter`.